autoscheduler/frontend/src/tests/AvailabilityUI.test.tsx

```diff
@@ -0,0 +1,267 @@
+import * as React from 'react';
+import {
+  render, fireEvent,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import autoSchedulerReducer from '../redux/reducers';
+import Schedule from '../components/Schedule/Schedule';
+import * as styles from '../components/Schedule/Schedule.css';
+
+describe('Availability UI', () => {
+  afterEach(() => {
+    document.body.innerHTML = '';
+  });
+
+  describe('adds availability cards', () => {
+    test('with a single click', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const {
+        getByText, getByLabelText, queryByText, queryByLabelText,
+      } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+      const mouseEventProps = {
+        button: 0,
+        clientY: 500,
+      };
+      // should create a time starting at mid-day, or 14:30
+      const expectedStart = '14:30';
+
+      // mocks height of the calendar day as 1000
+      const dayZero = document.getElementsByClassName(styles.calendarDay)[0];
+      jest.spyOn(dayZero, 'clientHeight', 'get')
+        .mockImplementation(() => 1000);
+
+      // act
+      const monday = getByLabelText('Monday');
+      fireEvent.mouseDown(monday, mouseEventProps);
+      fireEvent.mouseUp(monday, mouseEventProps);
+
+      // assert
+      expect(getByText('BUSY')).toBeInTheDocument();
+      // ensures that all times are actually calculated based on mocked measurements
+      expect(queryByText(/NaN/)).not.toBeInTheDocument();
+      expect(queryByLabelText('Adjust Start Time'))
+        .toHaveAttribute('aria-valuetext', expectedStart);
```

Comment:
Do you think it's worth asserting that the size of the availability component on click defaults to 30 minutes long, or is that more of a Redux/functional test, rather than a UI test?

Reply:
The 30-minute minimum is enforced by the UI, not Redux. Therefore, I will asjust the test to check that.